test/test_boundary_conditions_integration.jl

```diff
@@ -160,6 +164,69 @@ function test_flat_extrapolation_open_boundary_conditions(arch, FT)
     end
 end
 
+wall_normal_boundary_condition(::Val{1}, obc) = (; u = FieldBoundaryConditions(east = obc, west = obc))
+wall_normal_boundary_condition(::Val{2}, obc) = (; v = FieldBoundaryConditions(south = obc, north = obc))
+wall_normal_boundary_condition(::Val{3}, obc) = (; w = FieldBoundaryConditions(bottom = obc, top = obc))
+
+normal_velocity(::Val{1}, model) = model.velocities.u
+normal_velocity(::Val{2}, model) = model.velocities.v
+normal_velocity(::Val{3}, model) = model.velocities.w
+
+velocity_forcing(::Val{1}, forcing) = (; u = forcing)
+velocity_forcing(::Val{2}, forcing) = (; v = forcing)
+velocity_forcing(::Val{3}, forcing) = (; w = forcing)
+
+function test_pertubation_advection_open_boundary_conditions(arch, FT)
+    for orientation in 1:3
+        topology = tuple(map(n -> ifelse(n == orientation, Bounded, Flat), 1:3)...)
+
+        grid = RectilinearGrid(arch, FT; topology, size = (4, ), x = (0, 4), y = (0, 4), z = (0, 4), halo = (1, ))
+
+        obc = PerturbationAdvectionOpenBoundaryCondition(1, inflow_timescale = 0.1)
+
+        boundary_conditions = wall_normal_boundary_condition(Val(orientation), obc)
+
+        model = NonhydrostaticModel(; grid, boundary_conditions, timestepper = :QuasiAdamsBashforth2)
+
+        u = normal_velocity(Val(orientation), model)
+        
+        view(parent(u), :, :, :) .= 1
+
+        time_step!(model, 1)
+
+        @test all(view(parent(u), :, :, :) .== 1)
```

Comment:
Just some formatting suggestions to aid on readability. Kinda like separating the code into paragraphs that are on the same theme. I don't feel strongly about them so take it or leave it.

```suggestion
        u = normal_velocity(Val(orientation), model)
        view(parent(u), :, :, :) .= 1
        time_step!(model, 1)
        @test all(view(parent(u), :, :, :) .== 1)
```

Reply:
I like this better, I find double spaced code harder to read